src/applications/vre/definitions/profileAddress.js

```diff
@@ -0,0 +1,290 @@
+/**
+ * This uiSchema is modeled after how addresses are handled in the Profile application, and makes the same pattern
+ * available for use inside forms generated by the platform's form system.
+ */
+
+import React from 'react';
+import AdditionalInfo from '@department-of-veterans-affairs/formation-react/AdditionalInfo';
+import get from 'platform/utilities/data/get';
+import {
+  countries,
+  states50AndDC,
+  militaryCities,
+  militaryStates,
+} from 'vets-json-schema/dist/constants.json';
+
+/**
+ * CONSTANTS:
+ * 2. USA - references USA value and label
+ * 3. MilitaryBaseInfo - React component. Wrapped in AdditionalInfo component and used as description
+ */
+
+/**
+ * PATTERNS
+ * STREET_PATTERN - rejects white space only
+ * US_POSTAL_CODE_PATTERN - Matches 5 digit zipcodes and zip+4 patterns: 12345, 12345-1234
+ */
+const STREET_PATTERN = '^.*\\S.*';
+const US_POSTAL_CODE_PATTERN = '^\\d{5}(?:-\\d{4})?$';
+
+// Corresponding working form schema:
+export const addressSchema = {
+  type: 'object',
+  properties: {
+    isMilitary: {
+      type: 'boolean',
+    },
+    'view:livesOnMilitaryBaseInfo': {
+      type: 'object',
+      properties: {},
+    },
+    country: {
+      type: 'string',
+      enum: countries.map(country => country.value),
+      enumNames: countries.map(country => country.label),
+    },
+    addressLine1: {
```

Comment:
Prefill for addresses uses `street` and `street2` to prefill the address streets, I wonder if it would be better to use that since it would make prefill work out of the box without re-mapping anything? https://github.com/department-of-veterans-affairs/vets-api/blob/7360d520f2dfdd7b5a902139eaea44cef967b103/app/models/form_profile.rb#L51

Reply:
Oh good catch, I'll update. It brings up another question though - we offer 3 address lines, but only support prefill for two. Should a third street address line be added to prefill?

Reply:
I think we should support 3 address lines since vets profile has 3 lines. Not all endpoints accept 3 lines, so maybe in that case combine line 2 & 3 and truncate it, but that would really depend on the endpoint requirements.